server/src/session.ts

```diff
@@ -118,23 +139,23 @@ export class Session {
         this.logger.info(`Loading new project: ${event.data.reason}`);
         break;
       case ts.server.ProjectLoadingFinishEvent: {
-        const {project} = event.data;
-        try {
-          // Disable language service if project is not Angular
-          this.checkIsAngularProject(project);
-        } finally {
-          if (this.isProjectLoading) {
-            this.isProjectLoading = false;
-            this.connection.sendNotification(notification.ProjectLoadingFinish);
-          }
+        if (this.isProjectLoading) {
+          this.isProjectLoading = false;
+          this.connection.sendNotification(notification.ProjectLoadingFinish);
         }
+        this.checkProject(event.data.project);
         break;
       }
       case ts.server.ProjectsUpdatedInBackgroundEvent:
         // ProjectsUpdatedInBackgroundEvent is sent whenever diagnostics are
         // requested via project.refreshDiagnostics()
         this.triggerDiagnostics(event.data.openFiles);
         break;
+      case ts.server.ProjectLanguageServiceStateEvent:
+        this.connection.sendNotification(notification.ProjectLanguageService, {
+          projectName: event.data.project.getProjectName(),
+          languageServiceEnabled: event.data.languageServiceEnabled,
+        });
```

Comment:
The client doesn't actually use this right? This is just for the tests? (Making sure my mental model is correct)

Reply:
Yeah, it's just for tests in our setup. Although our vscode client doesn't do anything for this notification, I figured it's still useful to expose this for other clients. Notifications are cheap and non-blocking, so it's not too much overhead.